index.js

```diff
@@ -170,12 +167,10 @@ class KeyringController extends EventEmitter {
    * @param {string} password - The keyring controller password.
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
-  submitPassword(password) {
-    return this.unlockKeyrings(password).then((keyrings) => {
-      this.keyrings = keyrings;
-      this.setUnlocked();
-      return this.fullUpdate();
-    });
+  async submitPassword(password) {
+    this.keyrings = await this.unlockKeyrings(password);
+    this.setUnlocked();
+    this.fullUpdate(); // needs await?
```

Comment:
`fullUpdate` doesn't seem like an `async` method, so I don't think we need to await it.

Reply:
Good call -- sorry for leaving this in, was a TODO for me and I just forgot to remove it.